egi2bids/mff2bids.py

```diff
@@ -47,58 +57,115 @@
-    bids_path.update(session=session)
-    bids_path.update(task=task)
-    bids_path.update(datatype="eeg")
-    bids_path.update(run=run)
+    logger.info("Processing %s", mff_source)
+    working_dir = (
+        tempfile.TemporaryDirectory(suffix=".mff")
+        if working_dir is None
+        else working_dir
+    )
+    with working_dir as wd:
+        mff_source = _extract_folder(mff_source, dir_=wd)
 
-    print(Path(mff_source))
+    # BIDS root
+    bids_root = _ensure_path(bids_root, must_exist=True)
+    eeg_bids_path = BIDSPath(
+        root=bids_root,
+        subject=subject,
+        session=session,
+        task=task,
+        datatype="eeg",
+        run=run,
+    )
+    # JSON sidecar path
+    json_bids_path = eeg_bids_path.copy()
+    json_bids_path.update(extension=".json")
+    # Source path.
+    if save_source:
+        source_bids_root = bids_root.joinpath("sourcedata")
+        logger.info("Saving source data to %s", source_bids_root)
+        source_bids_path = eeg_bids_path.copy()
+        source_bids_path.update(root=source_bids_root)
+        source_bids_path = source_bids_path.fpath.with_suffix(
+            mff_source.suffix
+        )
+        if source_bids_path.exists() and overwrite is False:
+            raise ValueError(
+                f"Cannot write source data. Source data {source_bids_path}"
+                "already exists but overwrite is set to False."
+            )
 
     # load EEG data
-    raw = mne.io.read_raw_egi(Path(mff_source), preload=True)
-    raw.info["line_freq"] = line_frequency
+    raw = mne.io.read_raw_egi(mff_source, preload=True)
+    raw.info["line_freq"] = 50  # Hz, hard-coded for campus biotech/Europe.
 
     # rename channels
-    new_chs = {}
+    new_chs = {_extract_folder}
     for i, ch in enumerate(raw.info["ch_names"]):
         if i > 256:
             break
-        new_chs[ch] = ch_names_egi[i]
+        new_chs[ch] = CH_NAMES_EGI[i]
     raw.rename_channels(new_chs)
 
     # find events in stim channel
-    events_data = None
-    if event_id is not None:
-        events_data = mne.find_events(raw, stim_channel="STI 014")
-
-    # event_id = {"fixation cross":1,"stim":2}
-
-    # write BIDS
-    write_raw_bids(
-        raw,
-        bids_path,
-        format="BrainVision",
-        events_data=events_data,
-        event_id=event_id,
-        overwrite=True,
-        allow_preload=True,
-    )
-
-    bpath = bids_path.copy()
-    bpath.update(extension=".json")
+    stim_channel = "STI 014"
+    if stim_channel in raw.ch_names:
+        events_data = mne.find_events(raw, stim_channel=stim_channel)
+        # if event_id are not provided
+        if event_id is None:
+            event_ids = np.unique(events_data[:, 2])
+            event_id = {}
+            for event in event_ids:
+                event_id[f"Unkown_{event}"] = event
+        # TODO: check is provided events match events_data
+    else:
+        events_data = None
+        event_id = None
```

Comment:
I did not check this part, but the hard-coding of STI 014 is a bit weird. It seems like a MEGIN channel, not like an EGI one (IIRC it's the combine channel on older triux systems, now called STI101 and STI102). 
In the MFF files I shared with you, what is the name of the trigger channel? (If there is one).